components/scream/src/share/io/scream_scorpio_interface.F90

```diff
@@ -1449,9 +1450,13 @@ subroutine grid_read_darray_1d(filename, varname, var_data_ptr, time_index)
     var_size = SIZE(var%compdof)
 
     ! Now we know the exact size of the array, and can shape the f90 pointer
-    if (var%dtype .eq. pio_rtype) then
-      call c_f_pointer (var_data_ptr, var_data_real, [var_size])
-      call pio_read_darray(pio_atm_file%pioFileDesc, var%piovar, var%iodesc, var_data_real, ierr)
+    if (var%dtype .eq. PIO_DOUBLE) then
```

Comment:
I think we're already doing that a lot. Our input files have a certain precision (and some fields are double while other single), but we use it for both single and double precision testing. Someone mentioned (prob on slack) that PIO is allowing this to happen, so that the user can read in fp data without worrying about matching the precision of the input file.

Reply:
Correction: the issue is not reading a var with precision X when in the file it has precision Y. The issue was telling scorpio that we wanted to read/write a var with precision X, and then passing a pointer to Y. What's in the file is not the problem.